### PR TITLE
LAM-42 Ansible playbooks for http proxy, ssh bastion proxy, LAM-40 ansible inventory

### PR DESCRIPTION
**Pull request by ioantsaf**
#### LAM-40
- [x] Get okeanos output, assign hosts fqdn to dictionary
- [x] Make ansible inventory object using fqdn dictionary
- [x] Run test playbook within python, using inventory object
- [x] Add proxy_env host var
- [x] Add id host var
- [x] Add internal ip host var
- [x] Refactor input dictionary, as provisioner has new methods
- [x] Get subnet from provisioner and add as a host var for http proxy
- [x] Fix some ansible scripts variables
- [x] Refactor ansible scripts, so that common tasks run in parallel
- [x] Python tests
- [x] Readme
#### LAM-42
- [x] Install squid http proxy on master
- [x] Configure squid http proxy on master
- [x] Configure http_proxy environment var into ansible inventory python script
- [x] Add proxy_env environment into every task that uses get_url or apt modules
- [x] Make bastion ssh proxy, to run playbooks on slaves through master
- [x] Change /etc/hosts template. Now, internal ips correspond to the names snf-xxxxxx.local
- [x] Populate master public ssh key to slaves, via kamaki
- [x] Make python script that runs provisioner, then creates ansible inventory object and runs playbooks
- [x] Make playbook that waits for ssh demon to become ready, after cluster creation

### CODE REVIEW

**Comment by ioantsaf** [comment]
Please review @themiszamani @dpavlos @gouzouni625

File: ansible/README.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ There are four (4) roles and five (5) playbooks. These are:
 - apache-hadoop role, run from apache-hadoop playbook.
 - apache-kafka role, run from apache-kafka playbook.
 - apache-flink role, run from apache-flink playbook.
+- proxy role, run from proxy playbook.
 - cluster-install playbook which runs all the roles with the above sequence.
```
Comment (pkoro) [inline ansible/README.md:34]: Actual sequence is different that the order in readme file. Fix readme.

**Comment by pkoro** [comment]
@ioantsaf within `playbooks/` folder you have 5 subfolders with one playbook within each one. Here is what I see.

```
$ tree .
.
|-- apache-flink
|   `-- flink-install.yml
|-- apache-hadoop
|   `-- hadoop-install.yml
|-- apache-kafka
|   `-- kafka-install.yml
|-- cluster
|   `-- cluster-install.yml
|-- common
|   `-- install.yml
|-- proxy
|   `-- proxy.yml
|-- roles -> ../roles/
|-- testinventory.yml
|-- testproxy.yml
`-- wait_for_ssh.yml
```

I would suggest changing this to the structure below:

```
$ tree .
.
|-- apache-flink.yml
|-- apache-hadoop.yml
|-- apache-kafka.yml
|-- cluster-install.yml
|-- common.yml
|-- proxy.yml
|-- testinventory.yml
|-- testproxy.yml
`-- wait_for_ssh.yml
```

This way `1)` you wont need the explicit soft link for roles and `2)` when you reference a role within your playbook you just need only mention the name of the role. I.e. instead of 

```
    roles:
      - ../roles/proxy
```

you then just write:

```
    roles:
      - proxy
```

**Comment by gouzouni625** [comment]
+1

**Pull request merged**